### PR TITLE
RavenDB-14303 Cluster observer should ignore idle indexes on the prom…

### PR DESCRIPTION
**Pull request by efratshenhar**
…otable

### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-14303

### Additional description
Ignoring idle indexes on promotable node

### Type of change

- Bug fix

### How risky is the change?

- Low 

### Backward compatibility

- Non breaking change

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing 

- Tests have been added that prove the fix is effective or that the feature works

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Raven.Server/ServerWide/ServerStore.cs [hunk 1]
```diff
@@ -3137,6 +3137,7 @@ internal TestingStuff ForTestingPurposesOnly()
         internal class TestingStuff
         {
             internal Action BeforePutLicenseCommandHandledInOnValueChanged;
+            internal bool stopIndex;
```
Comment (ayende) [inline src/Raven.Server/ServerWide/ServerStore.cs]:
```suggestion
            internal bool StopIndex;
```

File: test/SlowTests/RavenDB-14303.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                {
+                    Name = "Sum",
+                };
+
+                var db = await dbServer.ServerStore.DatabasesLandlord.TryGetOrCreateResourceStore(databaseName);
+                var index = await db.IndexStore.CreateIndex(new AutoMapReduceIndexDefinition("Users", new[] { count, sum }, new[] { age }), Guid.NewGuid().ToString());
+
+                await leader.ServerStore.Engine.PutAsync(new SetIndexStateCommand(index.Name, IndexState.Idle, db.Name, Guid.NewGuid().ToString()));
+                dbServer = Servers.First(s => !createRes.NodesAddedTo.Contains(s.WebUrl));
+                dbServer.ServerStore.ForTestingPurposes = new ServerStore.TestingStuff {stopIndex = true};
```
Comment (ppekrol) [inline test/SlowTests/RavenDB-14303.cs]: there is a method called `ForTestingPurposesOnly` use it

File: test/SlowTests/RavenDB-14303.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+                await store.Maintenance.Server.SendAsync(new AddDatabaseNodeOperation(databaseName));
```
Comment (ppekrol) [inline test/SlowTests/RavenDB-14303.cs]: why it is important to have this 'StopIndex' set it TestingStuff?

Comment (efratshenhar) [reply test/SlowTests/RavenDB-14303.cs]: I wanted to see first that this test fail before the change so I needed the idle index to be stuck

File: test/SlowTests/RavenDB-14303.cs [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    {
+
+        [Fact]
+        public async Task IgnoreIdleIndexOnPromotable()
+        {
+            var clusterSize = 3;
+            var (_, leader) = await CreateRaftCluster(clusterSize);
+            var replicationFactor = 2;
+            var databaseName = GetDatabaseName();
+            using (var store = new DocumentStore()
```
Comment (karmeli87) [inline test/SlowTests/RavenDB-14303.cs]: use something like this:
```
using (GetDocumentStore(new Options
            {
                Server = cluster.Leader,
                ReplicationFactor = 2,

            }))
```

Comment (karmeli87) [reply test/SlowTests/RavenDB-14303.cs]: then you don't need stuff like `await dbServer.ServerStore.Cluster.WaitForIndexNotification(createRes.RaftCommandIndex);`

Comment (efratshenhar) [reply test/SlowTests/RavenDB-14303.cs]: done

File: test/SlowTests/RavenDB-14303.cs [hunk 5]
```diff
@@ -26,17 +26,18 @@ public async Task IgnoreIdleIndexOnPromotable()
...
+                ReplicationFactor = 2,
+
+            }))
             {
                 var doc = new DatabaseRecord(databaseName);
                 var createRes = await store.Maintenance.Server.SendAsync(new CreateDatabaseOperation(doc, replicationFactor));
 
                 var dbServer = Servers.First(s => createRes.NodesAddedTo.Contains(s.WebUrl));
-                await dbServer.ServerStore.Cluster.WaitForIndexNotification(createRes.RaftCommandIndex);
+
```
Comment (karmeli87) [inline test/SlowTests/RavenDB-14303.cs]: This all not need, right?

Comment (efratshenhar) [reply test/SlowTests/RavenDB-14303.cs]: yes

**Review by karmeli87 — APPROVED** [review]

**Pull request merged**